### PR TITLE
[RLlib] Add unit tests for updating episode data in base_env

### PR DESCRIPTION
**Pull request by mvindiola1**
## Why are these changes needed?

@sven1977, This is a follow up to issue #16683.

I had created a unit test for this issue but krfricke beat me to the PR so it did not make it. It is in this PR if you want it. It basically tests to make sure that the last_{obs,reward,done,info,action} are being set correctly for both single and multi-agent envs.

I also made one change to the base_env.py poll method. There is an extra clearing of self.last_info at the end of the method that I do not thin should be there. It is not there for any of the other last_x so I think it was put there incorrectly and should have been removed in yesterdays PR but was not.

## Checks

- [ x ] I've run `scripts/format.sh` to lint the changes in this PR.
- [N/A] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ x ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ x ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: rllib/evaluation/tests/test_episode.py [hunk 1]
```diff
@@ -0,0 +1,143 @@
+import ray
```
Comment (richardliaw) [inline rllib/evaluation/tests/test_episode.py:1]: nice! we just have to add this into `rllib/BUILD` to actually get this to run

File: rllib/env/base_env.py [hunk 2]
```diff
@@ -496,7 +496,6 @@ def poll(self) -> Tuple[MultiAgentDict, MultiAgentDict, MultiAgentDict,
                     del self.last_infos[ag]
 
         self.last_dones["__all__"] = False
-        self.last_infos = {}
```
Comment (richardliaw) [inline rllib/env/base_env.py:499]: what's the reason to remove this?

Comment (mvindiola1) [reply rllib/env/base_env.py:499]: @richardliaw,

To be consistent?  If you look at the whole method basically two things happen here wrt the last_* members. If the episode is done they are all dropped and replaced with empty dicts.  If the episode is not done then it releases only the values for specific agents in the observation.

Except for some reason right before the method returns it dropped the last_info dict and only the last_info the others (last_{obs, done, reward}) are left alone.

I figured this meant it was there on mistake.

Comment (richardliaw) [reply rllib/env/base_env.py:499]: No that's a good explanation! thanks for the thorough reply :)

File: rllib/evaluation/tests/test_episode.py (hunk 1 above)
Comment (mvindiola1) [reply rllib/evaluation/tests/test_episode.py:1]: @richardliaw It has been added to rllib/BUILD and is passing

**Comment by juliusfrost** [comment]
@mvindiola1 Could you resolve the merge conflict and formatting errors? Thanks!

**Comment by mvindiola1** [comment]
@juliusfrost, I just rebased and pushed. I am not sure what to do about the formatting issues. When I run format.sh it complains about two formatting issues. If I adjust those lines to try and fix the complaint then the next time I run format.sh it changes them back and complains again. ::shrug::

**Comment by juliusfrost** [comment]
@mvindiola1 Did you install the right linting dependencies? [`pip install -r python/requirements_linters.txt`](https://docs.ray.io/en/master/getting-involved.html#lint-and-formatting)

**Review by sven1977 — APPROVED** [review]
Awesome. Thanks @mvindiola1 for this test!

**Comment by sven1977** [comment]
Just waiting for LINT tests to pass now ...

**Comment by sven1977** [comment]
@mvindiola1 ^
Sorry for the delay!

**Pull request merged**